File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.js
function solution(s){
    var answer = true;
    let stk = [];
    for (k of s){
        if (k === '('){
            stk.push(k)
        }
        else {
            if (stk.length === 0) {
                return false
            }
            stk.pop()
        }
    }
    if (stk.length == 0){
        return true
    }
    else {
        return false
    }
}

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.py
from collections import deque
def solution(s):
    answer = True
    stk = deque()
    for i in range(len(s)):
        character = s[i]
        if not stk:
            stk.append(character)
            continue
        if character == "(":
            stk.append(character)
        else:
            if stk[-1] == "(":
                stk.pop()
    if stk:
        return False
    

    return True